src/main/kotlin/com/exactpro/th2/check1/rule/sequence/NoMessageCheckTask.kt

```diff
@@ -0,0 +1,117 @@
+import com.exactpro.th2.check1.rule.SailfishFilter
+import com.exactpro.th2.check1.util.VerificationUtil
+import com.exactpro.th2.check1.utils.fromProtoPreFilter
+import com.exactpro.th2.check1.utils.toRootMessageFilter
+import com.exactpro.th2.common.event.Event
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.RootMessageFilter
+import com.exactpro.th2.common.message.toJson
+import com.exactpro.th2.common.message.toTreeTable
+import com.exactpro.th2.common.schema.message.MessageRouter
+import io.reactivex.Observable
+import java.time.Instant
+
+class NoMessageCheckTask(
+    description: String?,
+    startTime: Instant,
+    sessionKey: SessionKey,
+    messageTimeout: Long? = null,
+    timeout: Long,
+    maxEventBatchContentSize: Int,
+    protoPreFilter: PreFilter,
+    parentEventID: EventID,
+    messageStream: Observable<StreamContainer>,
+    eventBatchRouter: MessageRouter<EventBatch>
+    ) : AbstractCheckTask(description, messageTimeout, timeout, maxEventBatchContentSize, startTime, sessionKey, parentEventID, messageStream, eventBatchRouter) {
+
+    private val protoPreMessageFilter: RootMessageFilter = protoPreFilter.toRootMessageFilter()
+    private val messagePreFilter = SailfishFilter(
+        converter.fromProtoPreFilter(protoPreMessageFilter),
+        protoPreMessageFilter.toCompareSettings()
+    )
+
+    private val metadataPreFilter: SailfishFilter? = protoPreMessageFilter.metadataFilterOrNull()?.let {
+        SailfishFilter(
+            converter.fromMetadataFilter(it, VerificationUtil.METADATA_MESSAGE_NAME),
+            it.toComparisonSettings()
+        )
+    }
+
+    private lateinit var preFilterEvent: Event
+    private lateinit var resultEvent: Event
+
+    private var preFilterMessagesCounter: Int = 0
+    private var extraMessagesCounter: Int = 0
+
+    init {
+        rootEvent
+            .name("No message check $sessionKey")
+            .type("noMessageCheck")
+    }
+
+    override fun onStart() {
+        super.onStart()
+        preFilterEvent = Event.start()
+            .type("preFiltering")
+            .bodyData(protoPreMessageFilter.toTreeTable())
+        rootEvent.addSubEvent(preFilterEvent)
+        resultEvent = Event.start()
+            .type("noMessagesCheckResult")
+        rootEvent.addSubEvent(resultEvent)
+    }
+
+    override fun onNext(messageContainer: MessageContainer) {
+        if (LOGGER.isDebugEnabled) {
+            LOGGER.debug("Received message with id: {}", messageContainer.protoMessage.metadata.id.toJson())
+        }
+
+        val result = matchFilter(
+            messageContainer,
+            messagePreFilter,
+            metadataPreFilter,
+            matchNames = false,
+            significant = false
+        )
+        val comparisonContainer = ComparisonContainer(messageContainer, protoPreMessageFilter, result)
+        if (comparisonContainer.fullyMatches) {
```

Comment:
If you only want to use `fullyMatches` you probably should move the extension method for AggregatedResult method from ComparisonContainer to some utility class. No needs to create an object just to access one method